packages/dd-trace/src/opentracing/tracer.js

```diff
@@ -105,13 +121,24 @@ class DatadogTracer {
       return null
     }
   }
+
+  _useTraceLevel (parent, options) {
+    // service trace level indicates service exit / entry spans only
+    if (!(SPAN_KIND in options.tags)) {
+      return new NoopSpan(this, parent, { keepParent: true })
+    }
+  }
 }
 
 function getContext (spanContext) {
   if (spanContext instanceof Span) {
     spanContext = spanContext.context()
   }
 
+  if (spanContext instanceof NoopSpan) {
+    spanContext = spanContext.context()
```

Comment:
This is the same code as the previous if-block so you can just put them together with `||`.

Reply:
fixed.